satellite/orders/orders_test.go

```diff
@@ -22,17 +22,64 @@ import (
 	"storj.io/storj/satellite"
 	"storj.io/storj/satellite/orders"
 	"storj.io/storj/satellite/satellitedb/satellitedbtest"
+	"storj.io/storj/storagenode"
 	"storj.io/storj/uplink"
 )
 
+func TestCleanArchive(t *testing.T) {
+	testplanet.Run(t, testplanet.Config{
+		SatelliteCount: 1, StorageNodeCount: 1, UplinkCount: 1,
+		Reconfigure: testplanet.Reconfigure{
+			StorageNode: func(index int, config *storagenode.Config) {
+				config.Storage2.Orders.CleanupInterval = 100 * time.Millisecond
+			},
+		},
+	}, func(t *testing.T, ctx *testcontext.Context, planet *testplanet.Planet) {
+		planet.Satellites[0].Audit.Service.Loop.Stop()
+		node := planet.StorageNodes[0]
+		service := node.Storage2.Orders
+		service.Sender.Pause()
+		service.Cleanup.Pause()
+
+		expectedData := testrand.Bytes(50 * memory.KiB)
+
+		redundancy := noLongTailRedundancy(planet)
+		err := planet.Uplinks[0].UploadWithConfig(ctx, planet.Satellites[0], &redundancy, "testbucket", "test/path", expectedData)
+		require.NoError(t, err)
+
+		service.Sender.TriggerWait()
+
+		// let archived order age for 100 milliseconds
+		time.Sleep(100 * time.Millisecond)
+
+		err = planet.Uplinks[0].UploadWithConfig(ctx, planet.Satellites[0], &redundancy, "testbucket", "test/path", expectedData)
+		require.NoError(t, err)
+
+		service.Sender.TriggerWait()
+
+		oldOrder, err := node.DB.Orders().ListArchived(ctx, 1000)
+		require.NoError(t, err)
+		require.Len(t, oldOrder, 2)
+
+		// trigger cleanup to delete archived orders older than 100 milliseconds
+		service.Cleanup.TriggerWait()
```

Comment:
Are we relying on this one not being deleted simply because we haven't waited 100 milliseconds yet? If so, this seems a high risk of being flappy, and I think we need to redesign this test somehow. Maybe a way to manually trigger orders being ready to archive.